c++/src/capnp/layout.c++

```diff
@@ -1609,31 +1636,61 @@ struct WireHelpers {
       return { segment, ptr };
     } else {
       // List of structs.
+      WordCount declDataSize = roundBitsUpToWords(value.structDataSize);
```

Comment:
Err, I realized I was a dummy after I posted that and deleted my comment before I saw your response to try to avoid bothering you. You are correct, and a DASSERT is fine.